engine/api/workflow/dao.go

```diff
@@ -918,7 +918,22 @@ func Update(ctx context.Context, db gorp.SqlExecutor, store cache.Store, proj sd
 
 	// Delete workflow data
 	if uptOption.OldWorkflow != nil && uptOption.OldWorkflow.ID != 0 {
-		if err := DeleteWorkflowData(db, *uptOption.OldWorkflow); err != nil {
+		if uptOption.OldWorkflow.WorkflowData == nil {
+			// this log should never happen
+			log.Error(ctx, "DeleteWorkflowData> workflowdata is nil on uptOption.OldWorkflow.ID:%d proj:%s", uptOption.OldWorkflow.ID, proj.Key)
+		}
+		// reload workflow
+		oldWf, err := LoadByID(ctx, db, store, proj, uptOption.OldWorkflow.ID, LoadOptions{
```

Comment:
DeepPipeline: false
Base64Keys: false
WithAsCodeUpdateEvent: false
WithIcon: false
WithIntegration: false 

no ?

Reply:
yes, only workflowData is used with this load